demo/src/main/java/com/orange/ods/demo/ui/MainScreen.kt

```diff
@@ -111,14 +132,52 @@ fun MainScreen(themeConfigurations: Set<OdsThemeConfigurationContract>) {
                         mainNavGraph(navigateToElement = mainState::navigateToElement)
                     }
                 }
+
+                if (changeThemeDialogVisible) {
+                    ChangeThemeDialog(themeState = mainState.themeState, onDismiss = {
+                        mainViewModel.storeUserThemeName(mainState.themeState.currentThemeConfiguration.name)
+                        changeThemeDialogVisible = false
+                    })
+                }
             }
         }
     }
 }
 
-private fun getCurrentThemeConfiguration(themeConfigurations: Set<OdsThemeConfigurationContract>): OdsThemeConfigurationContract {
-    // If another theme than Orange is injected, select it otherwise take the Orange theme which is always present
-    return themeConfigurations.firstOrNull { it.name != OrangeThemeConfiguration.OrangeThemeName }.orElse { themeConfigurations.first() }
+private fun getCurrentThemeConfiguration(storedUserThemeName: String?, themeConfigurations: Set<OdsThemeConfigurationContract>): OdsThemeConfigurationContract {
+    // Return the stored user theme configuration if it exists. If not, return the Orange theme configuration or the first existing theme configuration
+    return themeConfigurations.firstOrNull { it.name == storedUserThemeName }.takeIf { it != null }.orElse {
+        themeConfigurations.firstOrNull { it.name == OrangeThemeConfiguration.OrangeThemeName }.orElse { themeConfigurations.first() }
+    }
+}
+
+@Composable
+private fun ChangeThemeDialog(themeState: MainThemeState, onDismiss: () -> Unit) {
+    val selectedRadio = rememberSaveable { mutableStateOf(themeState.currentThemeConfiguration.name) }
+
+    Box(modifier = Modifier.padding(horizontal = dimensionResource(id = R.dimen.spacing_m))) {
```

Comment:
If I am correct, `Box` does not have any effect and could be removed.

Reply:
You're right, this was a test for dialog margin in Android 13.